crates/partition-store/src/snapshots/repository.rs

```diff
@@ -67,10 +67,26 @@ const MULTIPART_UPLOAD_CHUNK_SIZE_BYTES: usize = 5 * 1024 * 1024;
 /// Maximum number of concurrent downloads when getting snapshots from the repository.
 const DOWNLOAD_CONCURRENCY_LIMIT: usize = 8;
 
+/// Maximum number of pending deletions before snapshot creation is suspended. Deleting old
+/// snapshots is handled asynchronously from snapshot uploads as a follow-up metadata update. As
+/// pending deletes are thus persisted, they can be resumed by another process if the cleanup task
+/// fails. If for some reason delete operations are repeatedly failing, we can accumulate up to this
+/// many snapshots before we'll stop producing new snapshots.
+const MAX_PENDING_DELETIONS: usize = 10;
+
+#[derive(Debug, Clone, Copy, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub enum LatestSnapshotVersion {
+    #[default]
+    V1,
+    /// V2 adds support for a fixed number of retained snapshots
+    // todo(v1.7): make this the default
```

Comment:
Let's create a Github issue with the `release-blocker` label to help us remember this.